src/squidpy/im/__init__.py

```diff
@@ -3,7 +3,7 @@
 from __future__ import annotations
 
 from squidpy.im._container import ImageContainer
-from squidpy.im._feature import calculate_image_features
+from squidpy.im._feature import calculate_image_features, quantify_morphology
```

Comment:
hi @npeschke @timtreis was just taking a look at this, was wondering why the duplication of the function? It seems that `quantify_morphology` does something very similar to `calculate_image_features`. Do you plan to discontinue  the latter? wouldn't it be better to adapt the latter to use spatialdata + adapt? thank you

Reply:
tbh not sure yet what the optimal solution will be. There is clear redundancy and a strong overlap, but when I wrote the original MVP, it didn't clearly fit with the current `calculate_image_features`. On the one hand, some of these features don't need an image but only the label, but then also the structure of the function was going to be quite different.

Even if we merge them into the calculate_image_features function, the parameters of that function would have to change. So yeah, not fully sure yet.

Reply:
I generally wanted to have sth that takes an arbitrary callable with a region_props like footprint so we could also inject f.e. some HugginfFace featuriser or sth

Reply:
right, that's a good point. The `calculate_image_features` does support arbitrary functions, see https://squidpy.readthedocs.io/en/stable/api/squidpy.im.calculate_image_features.html and https://squidpy.readthedocs.io/en/stable/api/squidpy.im.calculate_image_features.html and I understand that the current implementation doesn't operate on masks (or not only on mask), but maybe it would be ok to just change the input to "image" to not be just an Image but also a Label?

Reply:
the parallelization/out of core functionality I think it's also important. The current implementation relies on joblib parallel to iterate over spots, and  understand it's not optimal (e.g. how to do it with raster labels and raster image?). But maybe a combination of that and dask, e.g. https://examples.dask.org/applications/image-processing.html could be useful. Basically I think we should strive to implement scalability in time/memory.

Reply:
looking at this, not maintained but maybe some ideas could be reused
https://github.com/jrussell25/dask-regionprops/blob/main/dask_regionprops/regionprops.py

Reply:
Yeah, this ties into the larger topic of moving things to the GPU

Reply:
Just for context, the current implementation takes 20 seconds on my machine to calculate all available features of the MIBI-TOF dataset. So parallelization might not need to be that urgent.

Reply:
It'll be if we add more features and analyse datasets like Xenium with 100k+ cells ;)

Reply:
hi @npeschke , thanks for sharing the time. If the mibi-tof dataset you are referring to is the one from squidpy, that is a toy dataset that does not really recapitulate real data complexity and size. I'd be curious to see the performance on a e.g. xenium dataset.